tools/proto2jsonschema/postgen.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const fs = require('fs');
+const recursive = require("recursive-readdir");
+const camelcaseKeys = require('camelcase-keys');
+
+/**
+ * This tool polishes the JSON schemas with a few modifications:
+ * - Adds "$id" – "." delimited ID – e.g. "google.events.cloud.audit.v1.LogEntryData"
```

Comment:
Minor (nit+1):
Top-level $id is meant to resolve to an absolute URI.
https://json-schema.org/understanding-json-schema/structuring.html#the-id-property
https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.8.2.2

This is technically valid, but it does lack the self-reference to canonical hosting URL which is often used and as of json-schema Draft 6 used as a base url for $ref.

Reply:
Changed to absolute URI ref per Chat discussion.